### PR TITLE
split openqa services to dependent systemd units

### PR DESCRIPTION
**Pull request by aaannz**
- this uses D-Bus **system bus**, since systemd service *type=dbus* requires it
- autostarting is not enabled here yet

### CODE REVIEW

File: lib/OpenQA/WebSockets/Server.pm [hunk 1]
```diff
@@ -208,8 +208,8 @@ sub setup {
 
     # use port one higher than WebAPI
     my $port = 9527;
-    if ($ENV{'MOJO_LISTEN'} =~ /.*:(\d{1,5})\/?$/) {
-        $port = $1 + 1;
+    if ($ENV{'MOJO_LISTEN'} && $ENV{'MOJO_LISTEN'} =~ /.*:(\d{1,5})\/?$/) {
```
Comment (coolo) [inline lib/OpenQA/WebSockets/Server.pm]: everyone loves the quotes :)

**Comment by aaannz** [comment]
- quotes removed (48d9a63)
- install service files (28834ad)

**Comment by aaannz** [comment]
packages for testing are at my [obs home](https://build.opensuse.org/package/show/home:oholecek:branches:devel:openQA/openQA)

**Comment by lnussel** [comment]
hmm, but now you are enforcing system bus which means it's no longer possible to run the webui as developer without installing and modifying the dbus config files, isn't it?

**Comment by aaannz** [comment]
I'm afraid so. Without adding org.opensuse.openqa DBus policy it won't allow to create DBus service. But hey, this is only RFC.

I can drop *type=dbus* from service files, we'll lose autostarting on dbus request but since I added scheduler and websockets as required for webui I don't think we really need it. The only disadvantage I can think of is when one is working on webui and is starting it manually, then autostarting may have some sense.

So possible scenarios I see:
1) accept this (require system dbus)
2) drop *type=dbus* and use user session bus (with the problems @lnussel mentioned on call - missing *$ENV{DISPLAY}* can cause trouble starting/connecting to user session bus
3) drop *type=dbus*, add custom dbus service as requirement and use custom dbus session (I didn't experiment with this very much so I don't know yet all the obstacles)

**Comment by lnussel** [comment]
Longer term I'd rather get rid of the system dbus usage.
Meanwhile could we tell the startup script via parameter or env variable whether to use system dbus? That way we could force it on the system bus only for the actual system service.

**Comment by aaannz** [comment]
If I remove d8595cb, then we'll continue use `Net::DBus->find` which checks for *$ENV{DBUS_STARTER_BUS_TYPE}* = `session` or `system` and by this it determine where to connect. I can make it like by default to connect to `session` (e.g. like Mojo by default starts in devel) and add *DBUS_STARTER_BUS_TYPE*=`system` to systemd service file as `Environment`.

This way we can keep *type=dbus*, enable autostarting and don't mess with DBus policy on dev machines. Will that work for you? One just needs to be aware when trying to start systemd service from git and not from package.

**Comment by aaannz** [comment]
pushed changes mentioned in previous comment except using `session` by default. `system` bus is now enforced when starting through systemd, when starting script directly I left it to `Net::DBus->find` to decide or user to manually specify env var.

**Comment by lnussel** [comment]
lgtm. with that we can make WebAPI OpenQA again

**Comment by aaannz** [comment]
By making WebAPI OpenQA again, do you mean to move lib/OpenQA/WebAPI/* back to lib/OpenQA?

**Comment by aaannz** [comment]
rebased and added missing 'Environment="DBUS_STARTER_BUS_TYPE=system"' to scheduler and websockets service files.

**Pull request merged**